internal/provider/resource_cloudflare_account.go

```diff
@@ -0,0 +1,150 @@
+package provider
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"log"
+
+	"github.com/MakeNowJust/heredoc/v2"
+	cloudflare "github.com/cloudflare/cloudflare-go"
+	"github.com/hashicorp/terraform-plugin-log/tflog"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+const (
+	accountTypeStandard   = "standard"
+	accountTypeEnterprise = "enterprise"
+)
+
+func resourceCloudflareAccount() *schema.Resource {
+	return &schema.Resource{
+		Schema:        resourceCloudflareAccountSchema(),
+		CreateContext: resourceCloudflareAccountCreate,
+		ReadContext:   resourceCloudflareAccountRead,
+		UpdateContext: resourceCloudflareAccountUpdate,
+		DeleteContext: resourceCloudflareAccountDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+		Description: heredoc.Doc(`
+			Provides a Cloudflare Account resource. Account is the basic resource for
+			working with Cloudflare zones, teams and users. Requires the tenant entitlement
+		`),
+	}
+}
+
+func resourceCloudflareAccountCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client := meta.(*cloudflare.API)
+	accountName := d.Get("name").(string)
+	var accountType string
+	if d.Get("type").(string) == "" {
+		accountType = accountTypeStandard
+	} else if d.Get("type").(string) == accountTypeStandard || d.Get("type").(string) == accountTypeEnterprise {
+		accountType = d.Get("type").(string)
+	} else {
+		return diag.FromErr(fmt.Errorf("invalid account type %s", d.Get("type").(string)))
+	}
+
+	tflog.Info(ctx, fmt.Sprintf("Creating Cloudflare Account: name %s", accountName))
+
+	account := cloudflare.Account{
+		Name: accountName,
+		Type: accountType,
+	}
+	acc, err := client.CreateAccount(ctx, account)
+
+	if err != nil {
+		return diag.FromErr(fmt.Errorf("error creating account %q: %w", accountName, err))
+	}
+
+	d.SetId(acc.ID)
+
+	return resourceCloudflareAccountRead(ctx, d, meta)
+}
+
+func getCloudflareAccontFromId(accountID string, client *cloudflare.API, ctx context.Context) (cloudflare.Account, error) {
```

Comment:
can you elaborate on why we need this method? we already have a library method to fetch by ID - https://github.com/cloudflare/cloudflare-go/blob/aaf504612ca6040b262b8867d21bd0f9c7ddc818/accounts.go#L75-L90

Reply:
no need to elaborate. Simply not used enough to your repo and missed this one. Just removed my redundant function.